### ISSUE TITLE
#️⃣ cannot be loaded

### ISSUE THREAD

**Issue by JamzumSum** [yaml/pyyaml#579]
~~~ py
> yaml.safe_load('1️⃣')
'1️⃣'
> yaml.safe_load('#️⃣')
None
~~~

This is tested under windows & linux, py39, `pyyaml=5.4.1` & `pyyaml=6.0`

So does anyone ever encounter this problem? Or if there is a solution for this?

**Comment by ingydotnet** [comment]
Emoji characters often have code points above x10FFFF which is the maximum YAML 1.2 supports.

See https://yaml.org/spec/1.2.2/#rule-c-printable

See https://www.quackit.com/character_sets/emoji/emoji_v3.0/unicode_emoji_v3.0_characters_all.cfm

The #️⃣ character doesn't even have a single code point. It has `&#x0023; &#xFE0F; &#x20E3;`.

x23 is the ascii `#` and the character that starts a YAML comment.

I believe pyyaml is seeing the ascii `#` and parsing your example as a comment.

In other words, the string `'#️⃣'` is not valid YAML 1.1 or 1.2.

We'll look into making it valid in 1.3 which is under design now.

**Comment by ingydotnet** [comment]
I was mistaken as I thought I saw code points in the page over `x10ffff`.
